sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/metrics/MetricsReader.java

```diff
@@ -70,55 +75,64 @@ public long getCounterMetric(String name, long defaultValue) {
     } catch (NoSuchElementException e) {
       LOG.error("Failed to get metric {}, from namespace {}", name, namespace);
     }
-    return defaultValue;
+    return ERRONEOUS_METRIC_VALUE;
   }
 
   /**
    * Return start time metric by counting the difference between "now" and min value from a
    * distribution metric.
    */
-  public long getStartTimeMetric(long now, String name) {
-    return this.getTimestampMetric(now, this.getDistributionMetric(name, DistributionType.MIN, -1));
+  public long getStartTimeMetric(String name) {
+    Iterable<MetricResult<DistributionResult>> timeDistributions = getDistributions(name);
+    return checkCredibility(getLowestMin(timeDistributions.iterator()));
```

Comment:
Would it be better for the `checkCredibility()` call be during the iteration inside `getLowestMin`? Perhaps a single min result should be discarded rather than all results.

Reply:
true - changing